lib/ember-orbit/cache.js

```diff
@@ -0,0 +1,49 @@
+const get = Ember.get;
+
+import LiveQuery from 'ember-orbit/live-query';
+import { parseIdentifier } from 'orbit-common/lib/identifiers';
+
+export default Ember.Object.extend({
+  _orbitCache: null,
+  _identityMap: null,
+
+  init(...args) {
+    this._super.apply(this, ...args);
+
+    Ember.assert(get(this, '_orbitCache'), '_orbitCache is required');
+    Ember.assert(get(this, '_identityMap'), '_identityMap is required');
+  },
+
+  retrieve(path) {
+    return get(this, '_orbitCache').get(path);
+  },
+
+  retrieveRecord(type, id) {
```

Comment:
Not sure what the semantics should be for methods that involve the identity map (`retrieveRecord` and `retrieveHasOne`). Because they retrieve records via the identity map a model is materialized even if the record hasn't been loaded yet. This makes sense for the model layer, but if a user is consuming the Cache api directly then perhaps they'd expect `undefined` for records that haven't been loaded. We could add an option to specify this behaviour but I wonder if we're missing a trick with the abstractions here i.e. perhaps the identity map should sit _between_ the Cache and the Model layer.